src/main/java/com/aws/greengrass/device/session/MqttSessionFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.device.session;
+
+import com.aws.greengrass.device.exception.AuthenticationException;
+import com.aws.greengrass.device.iot.Certificate;
+import com.aws.greengrass.device.iot.IotAuthClient;
+import com.aws.greengrass.device.iot.Thing;
+
+import java.util.Map;
+
+public class MqttSessionFactory implements SessionFactory {
```

Comment:
TODO: this factory needs to be registered

Reply:
How are the tests passing if this isn't registered? Does that imply that the integration tests have insufficient coverage?

Reply:
Only have unit tests at this point. Will add an integ test to cover this

Reply:
No doubt it'll be fine later, but I'm surprised that if this PR completely replaces the session management, that the existing integ tests (which should be exercising sessions) are still passing even when the MQTT factory isn't used. The existing integ tests are mocked or stubbed too much perhaps.